### DISCUSSION TITLE
Bugs with LSP-CSS and LESS files

### DISCUSSION THREAD

**Discussion by Amparose** [facelessuser/ColorHelper#263] [Q&A]
With only these packages installed:

```json
"installed_packages":
[
    "ColorHelper",
    "Less",
    "LSP",
    "LSP-css",
    "Package Control",
],
```

I get two colour hints in CSS files that open different modals:
![CleanShot 2024-04-12 at 14 43 01@2x](https://github.com/facelessuser/ColorHelper/assets/949192/85f6e0a7-0452-4ce6-8356-451416cbfb6d)

In .less files, only one of the hints (unfortunately the less useful one) appears:
![CleanShot 2024-04-12 at 14 53 55@2x](https://github.com/facelessuser/ColorHelper/assets/949192/c86ac483-fd6a-4c8f-8522-b2809a7697ec)

If I disable/remove LSP-CSS, then the hint and popup work as expected in a CSS file but not at all in a LESS file.

## Support Info
- ST ver.: 4169
- Platform: osx
- Arch: arm64
- Plugin ver.: 6.3.2
- Install via PC: True
- mdpopups ver.: 4.2.2

**Comment by facelessuser** [comment]
LSP provides its own color previews. If you only want one, you have to disable one. It seems you are here because you want ColorHelper previews, so I would check the LSP packages documentation or reach out the LSP-CSS author to find out how to disable the color previews.

This issue has arisen with other LSP providers, such as JSON. In that case, I tracked  down how to disable it. Maybe LSP-CSS is similar: https://github.com/facelessuser/ColorHelper/issues/190#issuecomment-913755314.

**Reply by facelessuser** [reply]
Additionally, if ColorHelper is not supported in Less files, we may have to include the syntax in our rules, but I have to know what syntax highlighter you are specifically using for Less.

**Reply by Amparose** [reply]
You are right: it's the same solution as the JSON issue linked. Add this to LSP-css.sublime-settings:

```json
"disabled_capabilities": {
    "colorProvider": true,
  },
```

ColorHelper is even mentioned in the default settings file comment! Thanks for helping track it down; I should have checked LSP settings first but it didn't occur to me.

This may be a separate issue but LESS variables that don't use hex values are missing the hint. This is why my LESS screenshot above only showed the LSP hint.

```less
@color1: #252525; // ColorHelper hint shows
@color2: rgb(37, 37, 37); // ColorHelper hint is missing
```

The syntax for LESS I am using is [this one](https://github.com/SublimeText/Less); could the lack of specific support you mention be the cause?

**Reply by facelessuser** [reply]
I can take a look. Syntax files get updated and sometimes change the scopes. Such changed can break us. Having people that use the syntax notifing us of the breakage helps us keep things up to date as I don't use all the various supported languages, just some of them
